router/batchrouter/asyncdestinationmanager/asyncdestinationmanager.go

```diff
@@ -22,8 +22,8 @@ import (
 
 type Asyncdestinationmanager interface {
 	Upload(destination *backendconfig.DestinationT, asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput
-	Poll(importingJob *jobsdb.JobT, payload []byte, timeout time.Duration) ([]byte, int)
-	FetchFailedEvents(*backendconfig.DestinationT, *utils.DestinationWithSources, []*jobsdb.JobT, *jobsdb.JobT, common.AsyncStatusResponse, time.Duration) ([]byte, int)
+	Poll(pollStruct common.AsyncPoll) ([]byte, int)
+	FetchFailedEvents(*utils.DestinationWithSources, []*jobsdb.JobT, *jobsdb.JobT, common.AsyncStatusResponse) ([]byte, int)
```

Comment:
this function signature also needs to be simplified

Reply:
changed this